phpseclib/Net/SFTP/Stream.php

```diff
@@ -315,16 +323,25 @@ private function _stream_read($count)
         //    return false;
         //}
 
-        $result = $this->sftp->get($this->path, false, $this->pos, $count);
-        if (isset($this->notification) && is_callable($this->notification)) {
-            if ($result === false) {
-                call_user_func($this->notification, STREAM_NOTIFY_FAILURE, STREAM_NOTIFY_SEVERITY_ERR, $this->sftp->getLastSFTPError(), NET_SFTP_OPEN, 0, 0);
-                return 0;
+        if (strlen($this->buffer) < $count) {
+            // read 80k chunks from sftp even though php will only request 8k chunk
+            // this allows us to minimize the overhead of making sftp requests
+            $chunk = $this->sftp->get($this->path, false, $this->pos, 81920);
```

Comment:
Can we make the 81920 a parameter? I think there is a mechanism for that on stream construction.

Reply:
I think what bantu is talking about is this:

```
<?php
require __DIR__ . '/vendor/autoload.php';

use phpseclib3\Net\SFTP\Stream;
use phpseclib3\Net\SFTP;

Stream::register();

$sftp = new SFTP('website.com');
$sftp->login('username', 'password');

$context = [
    'sftp' => ['sftp' => $sftp]
];
$context = stream_context_create($context);

$fp = fopen('sftp://dummy.com/home/vagrant/1mb', 'r', false, $context);
echo fread($fp, 10);
```
Other options are discussed at https://www.php.net/manual/en/wrappers.ssh2.php

Reply:
Yes, "stream context" is what I was referring to.